source/frontend/src/components/maps/leaflet/Control/AdvancedFilter/FilterContentContainer.tsx

```diff
@@ -12,42 +10,27 @@ export interface IFilterContentContainerProps {
 }
 
 export const FilterContentContainer: React.FC<IFilterContentContainerProps> = ({ View }) => {
-  const { isFiltering, setVisiblePimsProperties, setShowDisposed, setShowRetired, resetMapFilter } =
-    useMapStateMachine();
-
-  const { getMatchingProperties } = usePimsPropertyRepository();
-
-  const matchProperties = getMatchingProperties.execute;
-
-  const filterProperties = useCallback(
-    async (filter: Api_PropertyFilterCriteria) => {
-      const retrievedProperties = await matchProperties(filter);
-
-      if (retrievedProperties !== undefined) {
-        setVisiblePimsProperties(retrievedProperties);
-      }
-    },
-    [matchProperties, setVisiblePimsProperties],
-  );
+  const {
+    setShowDisposed,
+    setShowRetired,
+    resetMapFilter,
+    setAdvancedSearchCriteria,
+    isShowingMapFilter,
+    isLoading,
+  } = useMapStateMachine();
 
   const onChange = useCallback(
     async (model: PropertyFilterFormModel) => {
-      await filterProperties(model.toApi());
+      setAdvancedSearchCriteria(model);
       setShowDisposed(model.isDisposed);
       setShowRetired(model.isRetired);
     },
-    [filterProperties, setShowDisposed, setShowRetired],
+    [setShowDisposed, setShowRetired, setAdvancedSearchCriteria],
   );
 
   // Only render if the map state is filtering.
-  if (isFiltering) {
-    return (
-      <View
-        onChange={onChange}
-        onReset={resetMapFilter}
-        isLoading={getMatchingProperties.loading}
-      />
-    );
+  if (isShowingMapFilter) {
```

Comment:
It makes more sense to show this view based on this boolean, not based on whether an advanced filter is active.